File: InvoiceApi/ViewModels/InvoiceViewModel.cs
using InvoiceApi.Models;

namespace InvoiceApi.ViewModels
{
    public class InvoiceViewModel
    {
        public Guid Id { get; set; }

        public string InvoiceNumber { get; set; }

        public DateTime InvoiceDate { get; set; }

        public OrderViewModel Order { get; set; }

        public InvoiceState State { get; set; }

        public double TotalPrice { get; set; }
    }
}

File: invoice-fe/src/models/InvoiceViewModel.ts
import { OrderViewModel } from "./OrderViewModel";

export interface InvoiceViewModel {
    id: string;
    invoiceNumber: string;
    invoiceDate: Date;
    order: OrderViewModel | undefined;
    state: InvoiceState;
    totalPrice: number;
}

export enum InvoiceState {
  CREATED,
  SENT,
  CANCELLED,
  COMPLETED
}